framework/web/result.go

```diff
@@ -384,3 +429,114 @@ func (r *Responder) getLogger() flamingo.Logger {
 	}
 	return &flamingo.StdLogger{Logger: *log.New(os.Stdout, "flamingo", log.LstdFlags)}
 }
+
+//ApplyHeaders - sets the correct cache control headers
+func (c *CacheDirectives) ApplyHeaders(header http.Header) {
+	cacheControlValues := []string{}
+
+	if c.NoStore {
+		//No store makes all other headers obsolete
+		header.Set("Cache-Control", "no-store")
+		return
+	}
+
+	//Revalidation header:
+	if c.MustRevalidate {
+		cacheControlValues = append(cacheControlValues, "must-revalidate")
+	}
+	if c.ProxyRevalidate {
+		cacheControlValues = append(cacheControlValues, "proxy-revalidate")
+	}
+	if c.NoCache {
+		cacheControlValues = append(cacheControlValues, "no-cache")
+	} else {
+		if c.MaxAge > 0 {
+			header.Set("Expires", time.Now().Add(time.Duration(int64(c.MaxAge)) * time.Second).Local().Format(time.RFC1123))
+			cacheControlValues = append(cacheControlValues, fmt.Sprintf("max-age=%d", c.MaxAge))
+		}
+		if c.SMaxAge > 0 {
+			cacheControlValues = append(cacheControlValues, fmt.Sprintf("s-maxage=%d", c.SMaxAge))
+		}
+	}
+
+	//Add Validation Headers
+	if c.ETag != "" {
+		header.Set("ETag", c.ETag)
+	}
+	if c.LastModifiedSince != nil {
+		header.Set("Last-Modified", c.LastModifiedSince.Local().Format(time.RFC1123))
+	}
+
+	//Other directives for caches
+	if c.NoTransform {
+		cacheControlValues = append(cacheControlValues, "no-transform")
+	}
+	if c.Visibility == "public" {
+		cacheControlValues = append(cacheControlValues, "public")
+	}
+	if c.Visibility == "private" {
+		cacheControlValues = append(cacheControlValues, "private")
+	}
+
+	if len(cacheControlValues) > 0 {
+		header.Set("Cache-Control", strings.Join(cacheControlValues, ", "))
+	}
+}
+
+//NewCacheStrategy returns new cache Strategy
```

Comment:
This factory is not necessary, and actually not always used, e.g. by the Filter.